src/librustc_passes/ast_validation.rs

```diff
@@ -344,8 +344,11 @@ impl<'a> Visitor<'a> for AstValidator<'a> {
                                         trait_item.id, span,
                                         "patterns aren't allowed in methods without bodies");
                                 } else {
-                                    struct_span_err!(self.session, span, E0642,
-                                        "patterns aren't allowed in methods without bodies").emit();
+                                    let mut err = struct_span_err!(self.session, span, E0642,
+                                        "patterns aren't allowed in methods without bodies");
+                                    err.span_suggestion(span,
+                                        "use an underscore to ignore the name", "_".to_owned());
```

Comment:
What's the point of this recommendation?
If pattern doesn't work, then it's much more likely that the user will want some name describing the pattern as a whole, rather than no name at all.

Reply:
Synthesizing a meaningful name from code like `fn foo((x, y): (i32, i32));` can be impossible to get right (should it suggest `x_y`? `x_and_y`? something else?), which is why I suggested using `_`, but the wording could be changed slightly to `give this argument a name or use an underscore to ignore it, instead of a tuple pattern`.